### ISSUE TITLE
Knowledge Skillsoft Bug

### ISSUE THREAD

**Issue by goldeneye71** [chummer5a/chummer5a#2542]
(Please fill out the below as accurately as possible. More detail makes it easier for us to find the problems and fix them.)

### Environment
4.198,17

### Expected behaviour
deleting skillsofts deletes knowledge skill

### Actual behaviour
when knowsofts or linguasofts are added, it creates multiple copies within my knowledge section, and deleting the skillsoft does not remove the skill, which for languages, appear as native, rather than as their intended rank. 

### Steps to reproduce the behaviour
add skillsoft, delete skillsoft, check knowledge skills

### Link to affected character file
https://www.dropbox.com/s/40a96zs6sripf79/infil.chum5?dl=0

**Comment by DelnarErsike** [comment]
Odd... while I saw the duplicate knowledge skills on your character from what I assume were previous attempts at adding and removing skillsofts, when I tried to go in and reproduce it with a new knowsoft, the skill only got added once, and when I went to remove the knowsoft, the new skill properly got removed with it. Are you using any custom files perchance that affect knowsofts?

**Comment by goldeneye71** [comment]
Not that i know of, i only have the base chummer files and the nightly doem yesterday

**Comment by goldeneye71** [comment]
From yesterday*

**Comment by chummer5a** [comment]
This should be working correctly in current builds. At the very least I can't reproduce it with a new character file.

**Comment by Goberos** [comment]
I was able to figure out what was causing it and replicate it. 
I added Skilljack Rating 2
Added Linguasofts
If you then go back and change the Skilljack rating, the linguasofts on the skill page get copies that are locked in as Native.  Deleting the Skilljack will correct the problem... but even if you then add it back, the issue replicates automatically... as if the Languages are locked to the Skilljack cyberware.  

I deleted the Skilljack, converted the character to Career mode, and added the Skilljack for free, and the language section still populated with the locked in Native languages
